public/app/features/templating/template_srv.test.ts

```diff
@@ -95,6 +95,20 @@ describe('templateSrv', () => {
       expect(target).toBe('this.asd.filters');
     });
 
+    it('should replace ${test.name} with scoped text', () => {
+      const target = _templateSrv.replaceWithText('this.${test.name}.filters', {
+        test: { value: { name: 'mupp' }, text: 'asd' },
+      });
+      expect(target).toBe('this.mupp.filters');
+    });
+
+    it('should replace ${test.name} with variable text', () => {
+      const target = _templateSrv.replaceWithText('this.${test.name}.filters', {
+        test: { value: { name: 1 }, text: 'asd' },
+      });
+      expect(target).toBe('this.asd.filters');
+    });
```

Comment:
Not sure if this use case exists

Reply:
this looks wrong to me, value.name only has one representation here (name), the text is on a different level